gas/config/tc-riscv.c

```diff
@@ -142,17 +142,16 @@ riscv_multi_subset_supports (enum riscv_insn_class insn_class)
       return riscv_subset_supports ("f") && riscv_subset_supports ("c");
 
     case INSN_CLASS_Q: return riscv_subset_supports ("q");
-    case INSN_CLASS_V: return riscv_subset_supports ("v");
+
+    case INSN_CLASS_V:
+    case INSN_CLASS_V_AND_ZVLSSEG:
+      return riscv_subset_supports ("v");
```

Comment:
Maybe that should be
    case INSN_CLASS_V_AND_ZVLSSEG:
      return riscv_subset_supports ("v") || riscv_subset_supports ("zvlsseg");
which seems to be the intent of the V spec.  We probably don't have any way to enable zvlsseg without v but presumably this will be given a name later, and then when we add that this will work as intended.

And maybe the class name should be changed to INSN_CLASS_V_OR_ZVLSSEG?

Simiiarly with the zvamo support.

We can defer this to later if you think that is better though.

Reply:
How about simplify the Zv* class from INSN_CLASS_V_[AND|OR]_ZV* to INSN_CLASS_ZV*.  The naming INSN_CLASS_ZV* should clearly indicate that the instruction is a subset extension of V.  We only use the combination of naming when the instruction don't have a clear naming, like INSN_CLASS_F_AND_C and INSN_CLASS_V_AND_F, ...

One minor thing is that, are the zvamo and zvlsseg instructions possible to execute when we only enable the zvamo and zvlsseg subsets, but without enabling the base v?  Presumably we will have a Zvbase which means the instruction is belongs to base V, excepts zvamo and zvlsseg.  So we can change the code as follows,

case INSN_CLASS_V:
return riscv_subset_supports ("v")
or riscv_subset_supports ("zvbase");

case INSN_CLASS_ZVLSSEG:
return riscv_subset_supports ("v")
or (riscv_subset_supports ("zvbase") and riscv_subset_supports ("zvlsseg"));

case INSN_CLASS_ZVQMAC:
return (riscv_subset_supports ("v") and riscv_subset_supports ("zvqmac")
or (riscv_subset_supports ("zvbase") and riscv_subset_supports ("zvqmac"));